### PR TITLE
add ansi regr_count functions

### PR DESCRIPTION
**Pull request by yaooqinn**
Aiming to support all of the standard regr_xxxx functions, see SQL reference section 10.9.
 - regr_count
 - regr_slope
 - regr_intercept 
 - regr_r2
 - regr_sxx
 - regr_syy 
 - regr_sxy
 -  regr_avgx 
 - regr_avgy
 

Also, these functions are supported by all most all databases systems, e.g. snowflake, oracle, hive, db2, ....

### CODE REVIEW

**Review by MrPowers — APPROVED** [review]
@yaooqinn - LGTM, I like `org.apache.itachi.registerPostgresFunctions` 😉

File: docs/functions/postgres.md [hunk 1]
```diff
@@ -189,6 +189,37 @@ org.apache.spark.sql.catalyst.expressions.postgresql.JustifyDays
 
 - **Since**
 0.1.0
+## regr_count
+- **Usage**
+```scala
+
+    regr_count(expr1, expr2) - Returns the count of all rows in an expression pair. The function eliminates expression pairs where either expression in the pair is NULL.If no rows remain, the function returns 0.
```
Comment (MrPowers) [inline docs/functions/postgres.md:196]: Missing a space after NULL

File: src/main/scala/org/apache/spark/sql/catalyst/expressions/ansi/RegrCount.scala [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+
+
+object RegrCount {
+
+  val fd: FunctionDescription = (
+    new FunctionIdentifier("regr_count"),
+    ExpressionUtils.getExpressionInfo(classOf[RegrCount], "regr_count"),
+    (children: Seq[Expression]) => RegrCount(children.head, children.last))
+
+}
```
Comment (MrPowers) [inline src/main/scala/org/apache/spark/sql/catalyst/expressions/ansi/RegrCount.scala:84]: Missing newline

**Comment by MrPowers** [comment]
@yaooqinn - good job getting this merged 😄

**Comment by yaooqinn** [comment]
> @yaooqinn - good job getting this merged 😄

Thanks for the reviewing

**Pull request merged**